src/test/java/seedu/recipe/logic/commands/CommandTestUtil.java

```diff
@@ -20,18 +26,56 @@
-//                .withPhone(VALID_PHONE_BOB).withEmail(VALID_EMAIL_BOB).withAddress(VALID_ADDRESS_BOB)
-//                .withTags(VALID_TAG_HUSBAND, VALID_TAG_FRIEND).build();
-//    }
+    public static final String NAME_DESC_CHICKEN = " " + PREFIX_NAME + VALID_NAME_CHICKEN;
+    public static final String NAME_DESC_FISH = " " + PREFIX_NAME + VALID_NAME_FISH;
+    public static final String PORTION_DESC_CHICKEN = " " + PREFIX_PORTION + VALID_PORTION_CHICKEN;
+    public static final String PORTION_DESC_FISH = " " + PREFIX_PORTION + VALID_PORTION_FISH;
+    public static final String DURATION_DESC_CHICKEN = " " + PREFIX_DURATION + VALID_DURATION_CHICKEN;
+    public static final String DURATION_DESC_FISH = " " + PREFIX_DURATION + VALID_DURATION_FISH;
+    public static final String TAG_DESC_CHINESE = " " + PREFIX_TAG + VALID_TAG_CHINESE;
+    public static final String TAG_DESC_ITALIAN = " " + PREFIX_TAG + VALID_TAG_ITALIAN;
+    public static final String INGREDIENT_DESC_CHICKEN = " " + PREFIX_INGREDIENT + VALID_INGREDIENT_CHICKEN;
+    public static final String INGREDIENT_DESC_FISH = " " + PREFIX_INGREDIENT + VALID_INGREDIENT_FISH;
+    public static final String STEP_DESC_CHICKEN = " " + PREFIX_STEP + VALID_STEP_CHICKEN;
+    public static final String STEP_DESC_FISH = " " + PREFIX_STEP + VALID_STEP_FISH;
+
+    public static final String INVALID_NAME_DESC = " " + PREFIX_NAME + "Chicken&"; // '&' not allowed in names
+    public static final String INVALID_PORTION_DESC = " " + PREFIX_PORTION + "1 person"; // missing upper limit
+
+    public static final String INVALID_DURATION_DESC = " " + PREFIX_DURATION + "10"; // missing units
+    public static final String INVALID_TAG_DESC = " " + PREFIX_TAG + "Chinese*"; // '*' not allowed in tags
+    public static final String INVALID_INGREDIENT_DESC = " " + PREFIX_INGREDIENT + "2"; // number not allowed for ingredient
+    public static final String INVALID_STEP_DESC = " " + PREFIX_STEP + ""; // empty string not allowed for step
+
+    public static final String PREAMBLE_WHITESPACE = "\t  \r  \n";
+    public static final String PREAMBLE_NON_EMPTY = "NonEmptyPreamble";
+
+    public static final EditCommand.EditRecipeDescriptor DESC_CHICKEN;
+    public static final EditCommand.EditRecipeDescriptor DESC_FISH;
+
+    static {
+        DESC_CHICKEN = new EditRecipeDescriptorBuilder().withName(VALID_NAME_CHICKEN)
+                .withPortion(VALID_PORTION_CHICKEN).withDuration(VALID_DURATION_CHICKEN)
+                .withTags(VALID_TAG_CHINESE).withIngredients(VALID_INGREDIENT_CHICKEN)
+                .withSteps(VALID_STEP_CHICKEN).build();
+        DESC_FISH = new EditRecipeDescriptorBuilder().withName(VALID_NAME_FISH)
+                .withPortion(VALID_PORTION_FISH).withDuration(VALID_DURATION_FISH)
+                .withTags(VALID_TAG_ITALIAN).withIngredients(VALID_INGREDIENT_FISH)
+                .withSteps(VALID_STEP_FISH).build();
+    }
```

Comment:
just curious: does this have to be in a `static`-block?

i'm assuming directly defining `public static final EditCommand.EditRecipeDescriptor DESC_CHICKEN = new EditRecipeDescriptorBuilder()...` doesn't work?

Reply:
I haven't tried using this method yet cause this requires the edit command I think. I just reused the initial format for access modifier they gave which is static. I assume this is like protected in which we can use for the same package.